R/utils.R

```diff
@@ -191,7 +191,19 @@ check_filter_datanames <- function(filters, datanames) {
 teal_data_datanames <- function(data) {
   checkmate::assert_class(data, "teal_data")
   if (length(teal.data::datanames(data))) {
-    teal.data::datanames(data)
+    datanames <- teal.data::datanames(data)
+    joinkeys <- teal.data::join_keys(data)
+    if (length(join_keys) > 0) {
+      child_parent <- sapply(
+        datanames,
+        function(name) teal.data::parent(joinkeys, name),
+        USE.NAMES = TRUE,
+        simplify = FALSE
+      )
+      union(unlist(topological_sort(child_parent)), datanames)
```

Comment:
looks like this should be in `teal.data::datanames` **getter**. I think `topological_sort@669_insertUI@main` in `teal.data` is a better way to go - only tests would apply here

Reply:
If we want to include that in `teal.data` then I think we should adjust the code in all the pieces where
- datanames get changed
- join_keys get changed

As both of them influence the `topological_sort()`.

So I created a PR where we edited `teal_data` consturctor, `"datanames<-"` method and `"join_keys<-"` method. https://github.com/insightsengineering/teal.data/pull/318
However I will need to consult with Andre, if we need to edit `names(join_keys) <-` and `parents(join_keys) <- ` as well

Reply:
PR in `teal.data` https://github.com/insightsengineering/teal.data/pull/319 - will decrease this PR only to have tests